assets/js/modules/analytics-4/components/common/WebDataStreamSelect.js

```diff
@@ -0,0 +1,221 @@
+		if ( isDisabled ) {
+			return false;
+		}
+
+		const loadedAccounts =
+			select( MODULES_ANALYTICS ).hasFinishedResolution( 'getAccounts' );
+
+		const loadedProperties = select(
+			MODULES_ANALYTICS_4
+		).hasFinishedResolution( 'getProperties', [ accountID ] );
+
+		const loadedWebDataStreams = isValidPropertyID( propertyID )
+			? select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getWebDataStreams',
+					[ propertyID ]
+			  )
+			: true;
+
+		const finishedSelectingAccount =
+			select( MODULES_ANALYTICS ).hasFinishedSelectingAccount() !== false;
+
+		const isMatchingAccountProperty =
+			select( MODULES_ANALYTICS_4 ).isMatchingAccountProperty();
+
+		return (
+			isMatchingAccountProperty ||
+			! loadedAccounts ||
+			! loadedProperties ||
+			! loadedWebDataStreams ||
+			! finishedSelectingAccount
+		);
+	} );
+
+	const viewContext = useViewContext();
+	const { setWebDataStreamID, updateSettingsForMeasurementID } =
+		useDispatch( MODULES_ANALYTICS_4 );
+	const onWebDataStreamChange = useCallback(
+		( index, { dataset } ) => {
+			const newID = dataset.value;
+			if ( webDataStreamID === newID ) {
+				return;
+			}
+
+			setWebDataStreamID( newID );
+			updateSettingsForMeasurementID(
+				webDataStreams.find( ( { _id } ) => _id === newID )
+					?.webStreamData?.measurementId || '' // eslint-disable-line sitekit/acronym-case
+			);
+
+			trackEvent(
+				`${ viewContext }_analytics`,
+				newID === WEBDATASTREAM_CREATE
+					? 'change_webdatastream_new'
+					: 'change_webdatastream',
+				'ga4'
+			);
+		},
+		[
+			webDataStreams,
+			webDataStreamID,
+			setWebDataStreamID,
+			updateSettingsForMeasurementID,
+			viewContext,
+		]
+	);
+
+	if ( ! isValidAccountID( accountID ) ) {
+		return null;
+	} else if ( isLoading ) {
+		return <ProgressBar height={ 100 } small />;
+	}
+
+	const isValidSelection =
+		webDataStreamID === undefined || webDataStreamID === ''
+			? true
+			: isValidWebDataStreamSelection( webDataStreamID );
+
+	if ( hasModuleAccess === false ) {
+		return (
+			<Select
+				className={ classnames(
+					'googlesitekit-analytics-4__select-measurement',
+					className
+				) }
+				label={ __( 'Web Data Stream', 'google-site-kit' ) }
+				value={ measurementID }
+				enhanced
+				outlined
+				disabled
+			>
+				<Option value={ measurementID }>{ measurementID }</Option>
+			</Select>
+		);
+	}
+
+	return (
+		<Select
+			className={ classnames(
+				'googlesitekit-analytics-4__select-measurement',
+				className,
+				{
+					'mdc-select--invalid': ! isValidSelection,
+				}
+			) }
+			label={ __( 'Web Data Stream', 'google-site-kit' ) }
+			value={ webDataStreamID }
+			onEnhancedChange={ onWebDataStreamChange }
+			disabled={ isDisabled || ! isValidPropertySelection( propertyID ) }
+			enhanced
+			outlined
+		>
+			{ ( webDataStreams || [] )
+				.concat( {
+					_id: WEBDATASTREAM_CREATE,
+					displayName: __(
+						'Set up a new web data stream',
```

Comment:
We need to check with @marrrmarrr before we can go forward with the change in labelling of the field.

If we do, we'd also need to update this in the permissions message for the datapoint 
<img width="1049" alt="image" src="https://user-images.githubusercontent.com/1621608/225132244-d0cd0e45-110a-4611-96e1-17d2a021f9bc.png">

Reply:
The label is good as is, but we want to update the message in the permissions modal here for consistency. This has been updated in the AC as well.

Reply:
Yep, updated.